apps/sms/services/js/conversation/conversation_service.js

```diff
@@ -22,26 +38,74 @@ if (!('BridgeServiceMixin' in self)) {
   ]);
 
   var ConversationService = {
+    mobileMessageClient: null,
+
     /**
      * Initializes our service using the Service mixin's initService.
      */
     init() {
       this.initService();
+
+      this.mobileMessageClient = bridge.client(
+        'mozMobileMessageShim',
+        new BroadcastChannel('mobile-message-shim-channel')
+      ).plugin(streamClient);
     },
 
     /**
      * Stream that returns everything needed to display conversations in the
      * inbox view.
-     * @param {ServiceStream.<ConversationSummary>} The stream to use in the
-     * implementation: it will be passed autoamtically by the Bridge library.
+     * @param {ServiceStream.<ConversationSummary>} serviceStream The stream to
+     * use in the implementation: it will be passed automatically by the Bridge
+     * library.
      */
-    getAllConversations(stream) {},
+    getAllConversations(serviceStream) {
+      var getThreadsStream = this.mobileMessageClient.stream('getThreads');
```

Comment:
> Mmm, why do you think so? Should not client call abort on service stream when view is unloading (like we entered Inbox with lots of messages, navigating to Conversation, view knows that it's going to die so the client does and it should ask all active stream to be cancelled)? Otherwise our shared worker will be iterating through cursor even when user is in a different view. It's not implemented now, but do you think we won't need it?

Mmm right; wondering if this would be closed automatically if the page unload (because the shared worker port would be somehow dangling ?) Not sure.

Also I wonder what happens if the user enters a thread while the full thread list is not rendered yet, then we press back and we see the not-fully-rendered thread list using the BF Cache... (which doesn't work for us in Firefox now).